### ISSUE TITLE
Issue generating Secure API Key with SetRestrictIndices

### ISSUE THREAD

**Issue by NoelOConnell** [algolia/algoliasearch-client-csharp#140]
Hi,

I'm not sure if this is a bug or by design.
When trying to create a secure API key I would get an 'Null object reference' error.

```
var restrictIndices = new List<String>();            
restrictIndices.Add("index-name");

var filters = new Query().SetFilters($"field={someID}").SetRestrictIndices(restrictIndices);

String publicKey = client.GenerateSecuredApiKey(settings.APIKEY, filters);
```

I solved it by stepping through and seeing that GetQueryString was throwing the error when it was setting the Indices:
https://github.com/algolia/algoliasearch-client-csharp/blob/42ca1e7c549679fbc76d6c4217795f773d02ac75/Algolia.Search/Query.cs#L1293

`this.facets` was null and throwing the error.

I altered my query to include facets filter and it worked.

```
string[] facets = { "*" };

var filters = new Query().SetFilters($"field={someID}").SetFacets(facets).SetRestrictIndices(restrictIndices);
```
If you need more detail just let me know.

Thanks

**Comment by CBaptiste** [comment]
Hey @NoelOConnell,
Thx for reporting this, I am going to look into it, I ll come back to you asap !

Baptiste

**Comment by CBaptiste** [comment]
Indeed, there is an issue is the code you pointed, I am fixing it
Thx again

**Comment by NoelOConnell** [comment]
@CBaptiste Great! Glad I could help.

**Comment by CBaptiste** [comment]
Hey I just the fixed dll to nuget, 3.6.10
It should fix the issue,

Let me know !
